File: Assets/Scripts/Network/CommonStructures/NetworkPlayerInputState.cs
namespace Network
{
    /// <summary>
    /// The input state of a player to be sent over the network. Used to compute the simulation state.
    /// </summary>
    public class NetworkPlayerInputState
    {
        /// <summary>
        /// The simulation frame that this input state is for.
        /// </summary>
        public int simulationFrame;
        /// <summary>
        /// The direction that the player is facing.
        /// </summary>
        public float direction;
        /// <summary>
        /// The constructor for the NetworkPlayerInputState class.
        /// </summary>
        /// <param name="simulationFrame">The simulation frame that this input state is for.</param>
        /// <param name="direction">The direction that the player is facing.</param>
        public NetworkPlayerInputState(int simulationFrame, float direction)
        {
            this.simulationFrame = simulationFrame;
            this.direction = direction;
        }
    }
}

File: Server/src/commonStructures/networkPlayerInputState.ts
/**
 * The input state of a player to be sent over the network. Used to compute the simulation state.
 */
export default class NetworkPlayerInputState {
    /**
     * The simulation frame that this input state is for.
     */
    public simulationFrame:number;
    /**
     * The direction that the player is facing.
     */
    public direction:number;

    /**
     * The constructor for the NetworkPlayerInputState class.
     * @param simulationFrame The simulation frame that this input state is for.
     * @param direction The direction that the player is facing.
     */
    constructor(simulationFrame:number, direction:number) {
        this.simulationFrame = simulationFrame;
        this.direction = direction;
    }
}